### PR TITLE
chore(sdk): Add MaybeArbitrary as super trait

### LINKED ISSUE #12582: `MaybeArbitrary` super trait
### Describe the feature

Add `MaybeArbitrary` as super trait to `reth-primitives-traits` abstractions `Block`, `BlockHeader`, `BlockBody`, `TxType` and `Receipt`.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by zitup**
Closes https://github.com/paradigmxyz/reth/issues/12582.

### CODE REVIEW

File: crates/primitives/src/block.rs [hunk 1]
```diff
@@ -427,7 +427,7 @@ impl SealedBlock {
             return Err(GotExpected {
                 got: calculated_root,
                 expected: self.header.transactions_root,
-            })
+            });
```
Comment (emhane) [inline crates/primitives/src/block.rs:430]:
```suggestion
            })
```
pls revert unrelated formatting change (use `cargo +nightly fmt` instead of `cargo fmt`)

File: crates/primitives/src/block.rs [hunk 2]
```diff
@@ -467,6 +467,7 @@ impl<H, B> reth_primitives_traits::Block for SealedBlock<H, B>
 where
     H: reth_primitives_traits::BlockHeader + 'static,
     B: reth_primitives_traits::BlockBody + 'static,
+    SealedHeader<H>: MaybeArbitrary,
```
Comment (emhane) [inline crates/primitives/src/block.rs:470]: instead of this bound here, relax the arbitrary impl for `SealedHeader`
```rust
#[cfg(any(test, feature = "arbitrary"))]
impl<'a, H> arbitrary::Arbitrary<'a> for SealedHeader<H> where H: for<'a> arbitrary::Arbitrary<'a> {
    fn arbitrary(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
        let header = Header::arbitrary(u)?;

        Ok(Self::seal(header))
    }
}
```

**Review by emhane — CHANGES_REQUESTED** [review]
nice!

File: crates/primitives/src/block.rs (hunk 1 above)
Comment (zitup) [reply crates/primitives/src/block.rs]: Yeah, I have update the fmt setting of rust-analyzer :)

**Review by emhane — APPROVED** [review]

**Comment by zitup** [comment]
Some checks were not successful, should I do something?

There is an import order error in `primitives-traits/src/receipt.rs`,
`primitives-traits/src/header/sealed.rs` has an unused “MaybeArbitrary” error that I don't know why, `optimism/primitives/src/tx_type.rs` has an not found “arbitrary” error.

**Comment by emhane** [comment]
> Some checks were not successful, should I do something?
> 
> There is an import order error in `primitives-traits/src/receipt.rs`, `primitives-traits/src/header/sealed.rs` has an unused “MaybeArbitrary” error that I don't know why, `optimism/primitives/src/tx_type.rs` has an not found “arbitrary” error.

I will check it out

**Comment by emhane** [comment]
ok @mattsse , thanks for communicating
